### PR TITLE
Fix error prone warns

### PR DESCRIPTION
**Pull request by MaicolAntali**
I have fixed some other warns.

### CODE REVIEW

File: gson/src/main/java/com/google/gson/ExclusionStrategy.java [hunk 1]
```diff
@@ -93,12 +93,14 @@
 public interface ExclusionStrategy {
 
   /**
+   * Method use to decide if a field should be serialized or deserialized
```
Comment (eamonnmcmanus) [inline gson/src/main/java/com/google/gson/ExclusionStrategy.java:96]: I would write:

```java
 * Decides if a field should be skipped during serialization or deserialization.
 *
```

The main changes:
* blank line before the `@` tags;
* no need to say that this is a method, just what it does;
* explicitly talk about "skipping" since it is sort of the opposite of what you might expect from a method that is "deciding".

File: gson/src/main/java/com/google/gson/annotations/SerializedName.java [hunk 2]
```diff
@@ -83,10 +83,12 @@
 public @interface SerializedName {
 
   /**
+   * A String that indicates the desired name of the field when it is serialized or deserialized.
```
Comment (eamonnmcmanus) [inline gson/src/main/java/com/google/gson/annotations/SerializedName.java:86]: I'd go for brevity here too:

`The desired name of the field...`.

No need to say that it is a `String` since that is in the declared type.

File: gson/src/main/java/com/google/gson/annotations/SerializedName.java [hunk 3]
```diff
@@ -83,10 +83,12 @@
...
    * @return the desired name of the field when it is serialized or deserialized
    */
   String value();
   /**
+   * An array of strings that indicates the alternative names of the field when it is deserialized
```
Comment (eamonnmcmanus) [inline gson/src/main/java/com/google/gson/annotations/SerializedName.java:91]: Likewise:

`The alternative names...`

File: gson/src/main/java/com/google/gson/internal/ReflectionAccessFilterHelper.java [hunk 4]
```diff
@@ -98,6 +98,7 @@ private static abstract class AccessChecker {
             }
           };
         } catch (NoSuchMethodException ignored) {
+          // Ignored exception
```
Comment (eamonnmcmanus) [inline gson/src/main/java/com/google/gson/internal/ReflectionAccessFilterHelper.java:101]: I usually use something like `// OK: <reason why it is OK>`. Here, maybe:

`// OK: will assume everything is accessible`

Though I'm not sure it really _is_ OK. I think it might be better to wrap the exception in an `IllegalStateException` or something. But if we do that it should be in a separate PR.

File: gson/src/main/java/com/google/gson/internal/UnsafeAllocator.java [hunk 5]
```diff
@@ -64,6 +64,7 @@ public <T> T newInstance(Class<T> c) throws Exception {
         }
       };
     } catch (Exception ignored) {
+      // Ignored exception
```
Comment (eamonnmcmanus) [inline gson/src/main/java/com/google/gson/internal/UnsafeAllocator.java:67]: For these ones, I would write:

`// OK: try the next way`

File: gson/src/main/java/com/google/gson/internal/bind/DateTypeAdapter.java [hunk 6]
```diff
@@ -81,7 +81,9 @@ private Date deserializeToDate(JsonReader in) throws IOException {
       for (DateFormat dateFormat : dateFormats) {
         try {
           return dateFormat.parse(s);
-        } catch (ParseException ignored) {}
+        } catch (ParseException ignored) {
+          // Ignored exception
```
Comment (eamonnmcmanus) [inline gson/src/main/java/com/google/gson/internal/bind/DateTypeAdapter.java:85]: `// OK: try the next format`

File: gson/src/main/java/com/google/gson/internal/bind/DefaultDateTypeAdapter.java [hunk 7]
```diff
@@ -154,7 +154,9 @@ private Date deserializeToDate(JsonReader in) throws IOException {
       for (DateFormat dateFormat : dateFormats) {
         try {
           return dateFormat.parse(s);
-        } catch (ParseException ignored) {}
+        } catch (ParseException ignored) {
+          // Ignored exception
```
Comment (eamonnmcmanus) [inline gson/src/main/java/com/google/gson/internal/bind/DefaultDateTypeAdapter.java:158]: `// OK: try the next format`

**Review by eamonnmcmanus — CHANGES_REQUESTED** [review]
Thanks for doing this! Just some wording suggestions.

**Comment by MaicolAntali** [comment]
Thanks!

File: gson/src/main/java/com/google/gson/internal/JavaVersion.java [hunk 8]
```diff
@@ -75,13 +75,15 @@ private static int extractBeginningInt(String javaVersion) {
   }
 
   /**
+   * Gets the major Java version
```
Comment (eamonnmcmanus) [inline gson/src/main/java/com/google/gson/internal/JavaVersion.java:78]: Blank line before `@` tag. (I'm surprised Error Prone doesn't mention this.)

**Review by eamonnmcmanus — APPROVED** [review]
Thanks again!

**Review by haaami01 — COMMENTED** [review]
Thank You!!
Fixes the `EmptyCatch` warn given by ErrorProne
